object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -279,6 +304,313 @@ private static String getExactStringValue(Tree tree, ValueAnnotatedTypeFactory f
+      if (b != null) {
+        successorBlock.add((BlockImpl) b);
+      }
+    }
+    return successorBlock;
+  }
+
+  /**
+   * If the input block is Regular, the returned list of nodes is exactly the contents of the block.
+   * If it's an Exception_Block, then it returns the corresponding node that causes exception.
+   * Otherwise, it returns an emptyList.
+   *
+   * @param block
+   * @return List of Nodes
+   */
+  private List<Node> getBlockNodes(Block block) {
+    List<Node> blockNodes = new ArrayList<>();
+
+    switch (block.getType()) {
+      case REGULAR_BLOCK:
+        blockNodes = ((RegularBlockImpl) block).getContents();
+        break;
+
+      case EXCEPTION_BLOCK:
+        blockNodes.add(((ExceptionBlockImpl) block).getNode());
+    }
+    return blockNodes;
+  }
+
+  /**
+   * Updates {@code visited} and {@code worklist} if the input {@code state} has not been visited
+   * yet.
+   */
+  private void propagate(
+      BlockWithLocals state, Set<BlockWithLocals> visited, Deque<BlockWithLocals> worklist) {
+
+    if (!visited.stream()
+        .anyMatch(pair -> pair.block.equals(state.block) && pair.localSet.equals(state.localSet))) {
+      visited.add(state);
+      worklist.add(state);
+    }
+  }
+
+  /**
+   * Returns the String value of @AlwaysCall annotation declared on the class type of {@code
+   * element}. Returns null if the class type of {@code element} doesn't have @AlwaysCall
+   * annotation.
+   */
+  private @Nullable String getAlwaysCallValue(Element element) {
+
+    TypeMirror type = element.asType();
+    TypeElement eType = TypesUtils.getTypeElement(type);
+    AnnotationMirror alwaysCallAnnotation = getDeclAnnotation(eType, AlwaysCall.class);
+
+    return (alwaysCallAnnotation != null)
+        ? AnnotationUtils.getElementValue(alwaysCallAnnotation, "value", String.class, false)
+        : null;
+  }
+
+  /**
+   * Creates the appropriate @CalledMethods annotation that corresponds to the @AlwaysCall
+   * annotation declared on the class type of {@code assign.first}. Then, it gets @CalledMethod
+   * annotation of {@code assign.first} to do a subtyping check and reports an error if the check
+   * fails.
+   */
+  private void checkAlwaysCall(
+      Pair<LocalVariableNode, Tree> assign,
+      CFStore store,
+      AnnotatedTypeMirror annotatedTypeMirror) {
+
+    CFValue lhsCFValue = store.getValue(assign.first);
+    String alwaysCallValue = getAlwaysCallValue(assign.first.getElement());
+
+    boolean report = true;
+
+    if (lhsCFValue != null) { // When store contains the lhs
+      AnnotationMirror dummyCMAnno = createCalledMethods(alwaysCallValue);
+      AnnotatedTypeMirror annoType = getAnnotatedType(assign.first.getTree());
+      AnnotationMirror CMAnno = annoType.getAnnotationInHierarchy(TOP);
+      QualifierHierarchy qualifierHierarchy =
+          createQualifierHierarchy(new MultiGraphQualifierHierarchy.MultiGraphFactory(this));
```

Comment:
It shouldn't be necessary to create a new qualifier hierarchy - the type factory should already have one. You can use `AnnotatedTypeFactory#getQualifierHierarchy` instead.